### ISSUE TITLE
Investigate INT8 support for Torch MIGraphX

### ISSUE THREAD

**Issue by causten** [ROCm/AMDMIGraphX#1936]
Discover how Torch TensorRT does it
Check if models can be passed pre quantized

**Comment by shivadbhavsar** [comment]
Notes:

- TRT doesnt seem to have support for quantized layers (ie. modules under torch.nn.quantized.*) 
- Both dynamo and fx can trace pre-quantized models
- Quantization in torch can happen in statically and dynamically:
  - Dynamic pre-quantizes weights but activations are quantized on the fly (quantization params are also chosen during runtime it seems? https://github.com/pytorch/pytorch/blob/cc03e3a8922d286931bf5468e6dea9e82ff95625/aten/src/ATen/native/quantized/cpu/QuantUtils.h) 
    - Traceable by both fx and dynamo
    - Conv has no dynamic quantization
  - Static does both weights and activations (This is probably the one we want to focus on supporting
    - Traceable only via fx currently (dynamo errors out when dealing with quantized tensors)
- Summary of supported ops b/w static and dynamic: https://pytorch.org/docs/stable/quantization.html#quantization-api-summary

**Comment by shivadbhavsar** [comment]
Next Steps:
- Build simple converter for quantized (static) Linear and Conv layers to test functionality and accuracy
- Design how to handle quantized workflow if the initial tests work
